shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -940,6 +940,10 @@ public boolean performAction(
               virtualViewId, Action.DID_GAIN_ACCESSIBILITY_FOCUS);
           sendAccessibilityEvent(virtualViewId, AccessibilityEvent.TYPE_VIEW_ACCESSIBILITY_FOCUSED);
 
+          // It looks like sending the TYPE_VIEW_ACCESSIBILITY_FOCUSED event
+          // will trigger ACTION_CLEAR_ACCESSIBILITY_FOCUS, which sets
+          // accessibilityFocusedSemanticsNode to null before this code runs.
+          // TODO(goderbauer): Should we check if this is null before sending the TYPE_VIEW_ACCESSIBILITY_FOCUSED event?
```

Comment:
This comment isn't a fact, it brings more questions than answers questions.  I'd leave it out.

Reply:
Makes sense, I'm fine with leaving it out.

For context, I was trying to access the previously focused accessibility node (`accessibilityFocusedSemanticsNode`) here when prototyping a new feature, but I was finding that it would always be null (in my admittedly limited testing) due to the reason mentioned in my comment, which was not what I would have expected after reading this code right below:

```java
if (accessibilityFocusedSemanticsNode == null) {
    // When Android focuses a node, it doesn't invalidate the view.
    // (It does when it sends ACTION_CLEAR_ACCESSIBILITY_FOCUS, so
    // we only have to worry about this when the focused node is null.)
    rootAccessibilityView.invalidate();
}
accessibilityFocusedSemanticsNode = semanticsNode;
```

Is there a better way to document this phenomenon, or should we just not worry about it right now?

Reply:
It sounds like you are trying to say that the code should be changed.  I'd turn it into an issue.

Reply:
Good call. Documented the situation at https://github.com/flutter/flutter/issues/63823, feel free to edit and assign labels.

Most of the breadcrumbs from debugging these issues were actually documented in the framework with https://github.com/flutter/flutter/pull/63192, and now that the TODO is spun off into an issue, I am unsure if this engine PR has enough content to be worthwhile. I'd be fine closing it, but I'm curious about your thoughts.

Reply:
Given the issue has been filed, I'd just revert this one file. The rest looks good to merge to me.

Reply:
I've amended the patch and pushed an update.